### PR TITLE
feat(graphql): auto-generate <entity>_onchange mutations (Spec 64 §4.2, refs #148)

### PR DESCRIPTION
**Pull request by laofahai**
## Summary

Closes the M5 GraphQL gap left by PR #191/#198 — every entity that declares an `onchange` map now gets a `<name>_onchange` GraphQL mutation auto-generated alongside CRUD. The resolver mirrors the REST flow: CommandLayer with `skipActionSlots` runs auth/permission/tenant first, then OnchangeEvaluator returns `{ updates, warnings }`.

This is the last item in Spec 64 §13 M5; M6 (frontend AutoForm integration) remains a separate scope.

## Module

- **New module**: `addons/adapter-server/cap-adapter-server/src/graphql/build-onchange-mutations.ts` — keeps build-schema.ts size in check; one auto-generated mutation per entity with onchange, JSON-encoded \`updates\` + \`warnings: [String!]!\` response.
- **Schema integration**: \`BuildGraphQLSchemaOptions\` gains \`onchangeEvaluator?\`. When absent (or when CommandLayer is missing or has no permission middleware), the mutation is silently omitted — matches the REST 503 omission semantics.
- **Wiring**: \`dev.ts\` and \`capability.ts\` thread the evaluator into \`buildGraphQLSchema()\` so both the CLI dev server and the Spec 57 capability autoload path get the mutation.
- **Core change**: \`command-layer.ts\` — synthetic \`skipActionSlots\` result now includes \`actor: c.actor\` alongside \`tenantId\`/\`locale\` so the post-pipeline identity is observable to the caller (Spec 64 §9.1 — onchange runs with the caller's resolved permissions).
- **REST parity**: \`routes/onchange-api.ts\` — uses the new \`resolvedContext.actor\` so the REST path also benefits from auth-middleware actor enrichment (silently fixed alongside).

## Codex review — all 4 rounds resolved

| Round | Finding | Severity | Resolution |
|---|---|---|---|
| 1 | Non-auth pipeline failures rewritten to AUTHZ_DENIED (rate_limit/fail-closed misclassified) | P2 | Only 401/403 collapse to AUTHZ_DENIED; other codes preserved |
| 1 | Validation runs before auth — leaks endpoint existence | P3 | Reordered: CommandLayer first, then values validation |
| 1 | Unexpected evaluator errors echo \`err.message\` to client | P3 | Fixed message + log raw detail server-side |
| 2 | \`?? ctx.tenantId\` overrides middleware that intentionally clears tenant | P2 | Trust resolvedContext.tenantId verbatim |
| 3 | Pre-pipeline actor used (auth middleware enrichment ignored) | P2 | command-layer carries post-auth actor; both REST and GraphQL use it |
| 3 | \`internalSchemas\` filter ignored — internal schemas leak public mutation | P3 | Filter threaded through buildOnchangeMutationFields |
| 4 | Field name like \`sales-order_onchange\` crashes schema build | P2 | New \`sanitizeGraphQLFieldName()\` helper |
| 4 | Mutation advertised when no permission middleware (always fails at runtime) | P2 | Suppress at build time when \`getMiddlewares()\` lacks permission slot |
| 4 | Schema introspection enumerates onchange-enabled entities | P2 | Documented as system-wide limitation (orthogonal to onchange) |

## Gemini review

Round 5 — clean except 1 P3 nice-to-have (share \`MAX_VALUES_LENGTH\` constant with REST). Deferred: REST onchange has no body-size limit currently, so introducing one is a behavior change beyond this PR's scope.

## Test plan

- [x] \`bun run check\` — clean
- [x] \`bun run typecheck\` — clean
- [x] \`bun test\` — 4477 pass / 0 fail / 3 skip across 271 files
- [x] 20 new GraphQL onchange integration tests covering: auto-generation visibility, happy path with chained cascade, JSON validation (malformed/oversized/array), evaluator error code mapping, AUTHZ_DENIED canonicalization, validation-after-auth ordering, non-auth failure preservation, post-auth actor propagation, tenant scope clearing, internal schema filter, name sanitization with \`-\`, permission-middleware suppression, leaky-evaluator stub
- [x] 4 rounds of Codex cross-model review (round 5 hit usage limit; substituted with Gemini)
- [x] 1 round of Gemini cross-model review

Refs #148 (Phase 1 M5 GraphQL — closes the last item in Spec 64 §13 M5; M6 frontend AutoForm integration remains).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added GraphQL mutations to trigger onchange evaluators for entities with onchange configuration, returning structured updates and warnings.

* **Bug Fixes**
  * Improved authorization and error handling for onchange operations with consistent error codes and validation failure detection.

* **Tests**
  * Added comprehensive integration test suite validating schema behavior, successful evaluation, permission enforcement, error canonicalization, and edge cases.

### CODE REVIEW

File: addons/adapter-server/cap-adapter-server/src/graphql/build-onchange-mutations.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        // under the identity the auth slot produced (role hydration,
+        // impersonation) and under whatever scope the tenant slot
+        // resolved (or cleared). Falling back to the request-context
+        // values would silently undo middleware decisions and could
+        // either leak cross-tenant data or run lookups under broader
+        // permissions than the pipeline intended.
+        const resolvedContext =
+          commandResult.data && typeof commandResult.data === "object"
+            ? (commandResult.data as { actor?: Actor; tenantId?: string; locale?: string })
+            : undefined;
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/adapter-server/cap-adapter-server/src/graphql/build-onchange-mutations.ts]: Addressed in 5429d37 — extracted `extractSkipActionSlotsContext()` + `SkipActionSlotsResolvedContext` type into `routes/shared.ts`. Both REST onchange and GraphQL onchange resolver now use the shared helper, no runtime behavior change.
